libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -337,6 +338,21 @@ proc validateAndRelay(g: GossipSub,
     toSendPeers.excl(peer)
     toSendPeers.excl(seenPeers)
 
+    # IDontWant is only worth it if the message is substantially
+    # bigger than the messageId
+    if msg.data.len > msgId.len * 10:
+      g.broadcast(toSendPeers, RPCMsg(control: some(ControlMessage(
```

Comment:
we should only broadcast this to peers that haven't themselves sent us an idontwant for the given hash